x-pack/filebeat/input/http_endpoint/validate.go

```
@@ -50,5 +63,37 @@ func (v *apiValidator) ValidateHeader(r *http.Request) (int, error) {
 		return http.StatusUnsupportedMediaType, fmt.Errorf("Wrong Content-Type header, expecting %v", v.contentType)
 	}
 
+	if v.hmacHeader != "" && v.hmacKey != "" && v.hmacType != "" {
+		// We need access to the request body to validate the signature.
+		buf, _ := ioutil.ReadAll(r.Body)
+		rdr1 := ioutil.NopCloser(bytes.NewBuffer(buf))
+		originalBody := ioutil.NopCloser(bytes.NewBuffer(buf))
+
+		var bodyBytes, _ = ioutil.ReadAll(rdr1)
+		var mac hash.Hash
+		if v.hmacType == "sha256" {
+			mac = hmac.New(sha256.New, []byte(v.hmacKey))
+		} else {
+			mac = hmac.New(sha1.New, []byte(v.hmacKey))
+		}
+
+		mac.Write(bodyBytes)
+		actualMAC := mac.Sum(nil)
+
+		hmacHeaderValue := r.Header.Get(v.hmacHeader)
+		if v.hmacPrefix != "" {
+			hmacHeaderValue = strings.Replace(hmacHeaderValue, v.hmacPrefix, "", 1)
+		}
+
+		signature, _ := hex.DecodeString(hmacHeaderValue)
```

This should return an error.